### PR TITLE
1009 fix skipped tests

### LINKED ISSUE #1009: Fix temporarily skipped tests
```
── Skipped tests (3) ─────────────────────────────────────────────────────────────────────────────────────────────
• skipped until we resolve handling code in teal.data:::new_teal_data (2): test-module_nested_tabs.R:405:3,
  test-tdata.R:263:3
• todo: should we support data as unnamed list in teal? (1): test-init.R:17:3
```

### PR DESCRIPTION
**Pull request by chlebowa**
Fixes #1009

### CODE REVIEW

File: tests/testthat/test-tdata.R [hunk 1]
```diff
@@ -255,7 +255,6 @@ testthat::test_that("datasets are maintained during conversion", {
 
 testthat::test_that("code is maintained during conversion", {
```
Comment (averissimo) [inline tests/testthat/test-tdata.R:256]:
```suggestion
testthat::test_that("as_tdata maintains code during conversion", {
```

**Comment by chlebowa** [comment]
No more skipped tests.

**Review by pawelru — APPROVED** [review]
LGTM

**Review by averissimo — APPROVED** [review]
Looks good! I think it can be merged as is.

Feel free to accept or discard my only comment as you see best.

----

I did found an issue when inspecting the `teal.data` contents, as the hash of `data[["d2"]]` does not match the initial hash.

Issue created on `{teal.slice}` https://github.com/insightsengineering/teal.slice/issues/527

```r
rlang::hash(data[["d2"]])
#> [1] "fed25eb8e9c06544ff433b8cbb15379f"
```

**Comment by chlebowa** [comment]
> my only comment

I don't quite understand. This is not connected to this issue, is it?

**Comment by averissimo** [comment]
@chlebowa I was mentioning the suggestion that calculates the hash instead of using a manual string.

Doesn't matter now :-)

**Pull request merged**